tutorials/roofit/rf709_BarlowBeeston.py

```diff
@@ -0,0 +1,216 @@
+
+
+#  Fit all models to data and plot
+
+result0 = model0.fitTo(sumData, PrintLevel=0, Save=True)
+result1 = model1.fitTo(sumData, PrintLevel=0, Save=True)
+result2 = model2.fitTo(sumData, PrintLevel=0, Save=True)
+
+
+can = ROOT.TCanvas("can", "", 1500, 600)
+can.Divide(3, 1)
+
+pt = ROOT.TPaveText(-19.5, 1, -2, 25)
+pt.SetFillStyle(0)
+pt.SetBorderSize(0)
+
+
+can.cd(1)
+frame = x.frame(Title="No template uncertainties")
+# Plot data to enable automatic determination of model0 normalisation:
+sumData.plotOn(frame)
+model0.plotOn(frame, LineColor="b", VisualizeError=result0)
+# Plot data again to show it on top of model0 error bands:
+sumData.plotOn(frame)
+# Plot model components
+model0.plotOn(frame, LineColor="b")
+model0.paramOn(frame)
+
+sigData.plotOn(frame, MarkerColor="b")
+frame.Draw()
+
+pt_text1 = [
+    "No template uncertainties",
+    "are taken into account.",
+    "This leads to low errors",
+    "for the parameters A, since",
+    "the only source of errors",
+    "are the data statistics.",
+]
+for text in pt_text1:
+    pt.AddText(text)
+
+pt.DrawClone()
+
+
+can.cd(2)
+frame = x.frame(Title="Barlow Beeston for Sig & Bkg separately")
+sumData.plotOn(frame)
+model1.plotOn(frame, LineColor="b", VisualizeError=result1)
+# Plot data again to show it on top of error bands:
+sumData.plotOn(frame)
+model1.plotOn(frame, LineColor="b")
+model1.plotOn(frame, Components=p_ph_sig1, LineColor="kAzure")
+model1.plotOn(frame, Components=p_ph_bkg1, LineColor="r")
+model1.paramOn(frame, Parameters=ROOT.RooArgSet(Asig1, Abkg1))
+
+sigData.plotOn(frame, MarkerColor="b")
+frame.Draw()
+
+pt.Clear()
+
+pt_text2 = [
+    "With gamma parameters, the",
+    "signal & background templates",
+    "can adapt to the data.",
+    "Note how the blue signal",
+    "template changes its shape.",
+    "This leads to higher errors",
+    "of the scale parameters A.",
+]
+
+for text in pt_text2:
+    pt.AddText(text)
+
+pt.DrawClone()
+
+can.cd(3)
+frame = x.frame(Title="Barlow Beeston light for (Sig+Bkg)")
+sumData.plotOn(frame)
+model2.plotOn(frame, LineColor="b", VisualizeError=result2)
+# Plot data again to show it on top of model0 error bands:
+sumData.plotOn(frame)
+model2.plotOn(frame, LineColor="b")
+model2.plotOn(frame, Components=p_ph_sig2, LineColor="kAzure")
+model2.plotOn(frame, Components=p_ph_bkg2, LineColor="r")
+model2.paramOn(frame, Parameters=ROOT.RooArgSet(Asig2, Abkg2))
+
+sigData.plotOn(frame, MarkerColor="b")
+frame.Draw()
+
+pt.Clear()
+pt_text3 = [
+    "When signal and background",
+    "template share one gamma para-",
+    "meter per bin, they adapt less.",
+    "The errors of the A parameters",
+    "also shrink slightly.",
+]
+for text in pt_text3:
+    pt.AddText(text)
+
+pt.DrawClone()
+
+
+print("Asig [normal ] = " + Asig0.getVal() + " +/- " + Asig0.getError())
+print("\nAsig [BB     ] = " + Asig1.getVal() + " +/- " + Asig1.getError())
+print("\nAsig [BBlight] = " + Asig2.getVal() + " +/- " + Asig2.getError())
```

Comment:
You can't concatenate `str` with `double`, and there are also some additional newlines here that are not in the C++ version. What about:
```Python
print("Asig [normal ] = {} +/- {}".format(Asig0.getVal(), Asig0.getError()))
print("Asig [BB     ] = {} +/- {}".format(Asig1.getVal(), Asig1.getError()))
print("Asig [BBlight] = {} +/- {}".format(Asig2.getVal(), Asig2.getError()))
```